src/widgets/scrollbar.rs

```diff
@@ -1128,4 +1131,341 @@ mod tests {
             assert_buffer_eq!(buffer, Buffer::with_lines(expected.clone()));
         }
     }
+
+    #[rstest]
+    #[case("██", 0, 2, "position_0")]
+    #[case("██", 1, 2, "position_1")]
+    fn render_scrollbar_simplest(
+        #[case] expected: &str,
+        #[case] position: usize,
+        #[case] content_length: usize,
+        #[case] assertion_message: &str,
+    ) {
+        let size = expected.width() as u16;
+        let mut buffer = Buffer::empty(Rect::new(0, 0, size, 1));
+        let mut state = ScrollbarState::default()
+            .position(position)
+            .content_length(content_length);
+        Scrollbar::default()
+            .orientation(ScrollbarOrientation::HorizontalBottom)
+            .begin_symbol(None)
+            .end_symbol(None)
+            .render(buffer.area, &mut buffer, &mut state);
+        assert_eq!(
+            buffer,
+            Buffer::with_lines(vec![expected]),
+            "{}",
+            assertion_message
+        );
+    }
+
+    #[rstest]
+    #[case("##########", 0, 10, "position_0")]
+    #[case("##########", 1, 10, "position_1")]
+    #[case("##########", 2, 10, "position_2")]
+    #[case("##########", 3, 10, "position_3")]
+    #[case("##########", 4, 10, "position_4")]
+    #[case("##########", 5, 10, "position_5")]
+    #[case("##########", 6, 10, "position_6")]
+    #[case("##########", 7, 10, "position_7")]
+    #[case("##########", 8, 10, "position_8")]
+    #[case("##########", 9, 10, "position_9")]
+    fn render_scrollbar_simple(
+        #[case] expected: &str,
+        #[case] position: usize,
+        #[case] content_length: usize,
+        #[case] assertion_message: &str,
+    ) {
+        let size = expected.width();
+        let mut buffer = Buffer::empty(Rect::new(0, 0, size as u16, 1));
+        let mut state = ScrollbarState::default()
+            .position(position)
+            .content_length(content_length);
+        Scrollbar::default()
+            .orientation(ScrollbarOrientation::HorizontalTop)
+            .begin_symbol(None)
+            .end_symbol(None)
+            .track_symbol(Some("-"))
+            .thumb_symbol("#")
+            .render(buffer.area, &mut buffer, &mut state);
+        assert_eq!(
+            buffer,
+            Buffer::with_lines(vec![expected]),
+            "{}",
+            assertion_message,
+        );
+    }
+
+    #[rstest]
+    #[case("          ", 0, 0, "position_0")]
+    fn render_scrollbar_nobar(
```

Comment:
~~Is there any point to this test? Testing that is renders empty instead of the block border?~~

Nit but it confused me, really this tests an `empty` bar and not an invisible bar (with no track or thumb).

Reply:
+1 comment / more tests around this.